### DISCUSSION TITLE
Should fillComplete() only be called from the root process?

### DISCUSSION THREAD

**Discussion by abodado** [trilinos/Trilinos#13499] [Q&A] [answered]
Quick question... (@tpetra, @csiefer2), 
When doing fillComplete() is it required to only call fillComplete() from the root process (when using MPI)? 
for example:

```
RCP< const Teuchos::Comm<int> > comm = Teuchos::DefaultComm<int>::getComm();
int rank   = comm->getRank();

// do matrix fill

if (rank==0) {
  matrixA->fillComplete();
}
```

**Comment by cgcgcg** [comment, marked as answer]
No, it should be called on every rank in the communicator.

**Reply by abodado** [reply]
ok, thank you!
